compiler-core/src/analyse.rs

```diff
@@ -1343,3 +1343,72 @@ fn get_compatible_record_fields<A>(
 
     compatible
 }
+
+/// Given a type, return a list of all the types it depends on
+fn get_type_dependencies(type_name: SmolStr, typ: &TypeAst) -> Vec<SmolStr> {
+    let mut deps = Vec::with_capacity(1);
+
+    match typ {
+        TypeAst::Var { name, .. } => deps.push(name.clone()),
+        TypeAst::Hole { name, .. } => deps.push(name.clone()),
```

Comment:
These are not dependencies, we don't need to collect them.
```suggestion
        TypeAst::Var { name, .. } =>(),
        TypeAst::Hole { name, .. } => (),
```